azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebAppBase.java

```diff
@@ -0,0 +1,792 @@
+     */
+    PythonVersion pythonVersion();
+
+    /**
+     * @return the version of Node.JS
+     */
+    String nodeVersion();
+
+    /**
+     * @return if the remote eebugging is enabled
+     */
+    boolean remoteDebuggingEnabled();
+
+    /**
+     * @return the remote debugging version
+     */
+    RemoteVisualStudioVersion remoteDebuggingVersion();
+
+    /**
+     * @return if web socket is enabled
+     */
+    boolean webSocketsEnabled();
+
+    /**
+     * @return if the web app is always on
+     */
+    boolean alwaysOn();
+
+    /**
+     * @return Java version
+     */
+    JavaVersion javaVersion();
+
+    /**
+     * @return Java container
+     */
+    String javaContainer();
+
+    /**
+     * @return Java container version
+     */
+    String javaContainerVersion();
+
+    /**
+     * @return managed pipeline mode
+     */
+    ManagedPipelineMode managedPipelineMode();
+
+    /**
+     * @return the auto swap slot name
+     */
+    String autoSwapSlotName();
+
+    /**
+     * Start the web app or deployment slot.
+     */
+    void start();
+
+    /**
+     * Stop the web app or deployment slot.
+     */
+    void stop();
+
+    /**
+     * Restart the web app or deployment slot.
+     */
+    void restart();
+
+    /**************************************************************
+     * Fluent interfaces to provision a Web App or deployment slot
+     **************************************************************/
+
+    /**
+     * Container interface for all the definitions that need to be implemented.
+     */
+    interface Definition<FluentT> extends
+            DefinitionStages.WithAppServicePlan<FluentT>,
+            DefinitionStages.WithHostNameBinding<FluentT>,
+            DefinitionStages.WithHostNameSslBinding<FluentT>,
+            DefinitionStages.WithWebContainer<FluentT>,
+            DefinitionStages.WithCreate<FluentT> {
+    }
+
+    /**
+     * Grouping of all the site definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * A web app definition allowing app service plan to be set.
+         * @param <FluentT> the type of the resource, either a web app or a deployment slot
+         */
+        interface WithAppServicePlan<FluentT> {
+            /**
+             * Create a new free app service plan to use. No custom domains or SSL bindings are available in this plan.
+             * @return the next stage of web app definition
+             */
+            WithCreate<FluentT> withNewFreeAppServicePlan();
+
+            /**
+             * Create a new app service plan to use.
+             * @param name the name of the app service plan
+             * @param pricingTier the pricing tier to use
+             * @return the next stage of web app definition
+             */
+            WithHostNameBinding<FluentT> withNewAppServicePlan(String name, AppServicePricingTier pricingTier);
+
+            /**
+             * Use an existing app service plan for the web app.
```

Comment:
"Uses.."
I won't comment on more, just recommend re-reading all javadocs and correcting to 3rd party declarative mood.